File: app/src/main/java/com/tangem/presentation/dialog/NoExtendedLengthSupportDialog.java
package com.tangem.presentation.dialog;

import android.app.AlertDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.content.DialogInterface;
import android.os.Bundle;

import com.tangem.wallet.R;

public class NoExtendedLengthSupportDialog extends DialogFragment {
    public static final String TAG = NoExtendedLengthSupportDialog.class.getSimpleName();

    public static boolean allReadyShowed = false;

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        return new AlertDialog.Builder(getActivity())
                .setIcon(R.drawable.tangem_logo_small_new)
                .setTitle(R.string.warning)
                .setMessage(R.string.the_nfc_adapter_length_apdu)
                .setPositiveButton(R.string.got_it, (dialog, whichButton) -> NoExtendedLengthSupportDialog.allReadyShowed = true)
                .create();
    }

    @Override
    public void onCancel(DialogInterface dialog) {
        super.onCancel(dialog);
    }

}

File: app/src/main/java/com/tangem/presentation/dialog/NoExtendedLengthSupportDialog.kt
package com.tangem.presentation.dialog

import android.app.AlertDialog
import android.app.Dialog
import android.app.DialogFragment
import android.os.Bundle

import com.tangem.wallet.R

class NoExtendedLengthSupportDialog : DialogFragment() {

    companion object {
        val TAG: String = NoExtendedLengthSupportDialog::class.java.simpleName
        var allReadyShowed = false
        var message = ""// = R.string.the_nfc_adapter_length_apdu
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return AlertDialog.Builder(activity)
                .setIcon(R.drawable.tangem_logo_small_new)
                .setTitle(R.string.warning)
                .setMessage(message)
                .setPositiveButton(R.string.got_it) { _, _ -> NoExtendedLengthSupportDialog.allReadyShowed = false }
                .create()
    }

}